controllers/pod_reconciler.go

```diff
@@ -240,212 +232,38 @@ func inspectImages(ctx context.Context, clientset *kubernetes.Clientset, pod *co
-	authFile, err := WriteAuthFile(&secretList, pod.Name+"-"+pod.Namespace)
-	if err != nil {
-		klog.Warningf("Couldn't write auth file for %s ns: %s %v", pod.Name, pod.Namespace, err)
-		return nil, err
-	} else {
-		defer func(f *os.File) {
-			if err := f.Close(); err != nil {
-				klog.Warningf("Failed to close auth file %s %v", f.Name(), err)
-			}
-		}(authFile)
-	}
-
-	src, err := ref.NewImageSource(ctx, &types.SystemContext{
-		AuthFilePath: authFile.Name(),
-	})
-	if err != nil {
-		klog.Warningf("Error creating the image source: %v", err)
-		return nil, err
-	}
-	defer func(src types.ImageSource) {
-		err := src.Close()
-		if err != nil {
-			klog.Warningf("Error closing the image source for the %s/%s's image %s: %v",
-				pod.Namespace, pod.Name, imageName, err)
-		}
-	}(src)
-
-	rawManifest, _, err := src.GetManifest(ctx, nil)
-	if err != nil {
-		klog.Infof("Error getting the image manifest: %v", err)
-		return nil, err
-	}
-	if manifest.MIMETypeIsMultiImage(manifest.GuessMIMEType(rawManifest)) {
-		klog.V(5).Infof("%s/%s's image %s is a manifest list... getting the list of supported architectures",
-			pod.Namespace, pod.Name, imageName)
-		// The image is a manifest list
-		index, err := manifest.OCI1IndexFromManifest(rawManifest)
-		if err != nil {
-			klog.Warningf("Error parsing the OCI index from the raw manifest of the %s/%s's image %s: %v",
-				pod.Namespace, pod.Name, imageName, err)
-		}
-		supportedArchitectures = sets.New[string]()
-		for _, m := range index.Manifests {
-			supportedArchitectures = sets.Insert(supportedArchitectures, m.Platform.Architecture)
-		}
-		return supportedArchitectures, nil
-
-	} else {
-		klog.V(5).Infof("%s/%s's image %s is not a manifest list... getting the supported architecture",
-			pod.Namespace, pod.Name, imageName)
-		sys := &types.SystemContext{}
-		parsedImage, err := image.FromUnparsedImage(ctx, sys, image.UnparsedInstance(src, nil))
-		if err != nil {
-			klog.Warningf("Error parsing the manifest of the %s/%s's image %s: %v",
-				pod.Namespace, pod.Name, imageName, err)
-			return nil, err
-		}
-		config, err := parsedImage.OCIConfig(ctx)
-		if err != nil {
-			// Ignore errors due to invalid images at this stage
-			klog.Warningf("Error parsing the OCI config of the %s/%s's image %s: %v",
-				pod.Namespace, pod.Name, imageName, err)
-			return nil, err
-		}
-		return sets.New(config.Architecture), nil
-	}
-}
-
-// Function consolidates image pull secrets
-// - ImagePullSecrets in pod
-// - global pull secret in openshift-config
-// TODO? - default pull secret in pod namespace if no ImagePullSecrets field?
-func PullSecretAuthList(ctx context.Context, clientset *kubernetes.Clientset, pod *corev1.Pod) (RegAuthList, error) {
-	secretAuths := make(RegAuthList)
+// pullSecretAuthList returns the list of secrets data for the given pod given its imagePullSecrets field
+func pullSecretAuthList(ctx context.Context, clientset *kubernetes.Clientset, pod *corev1.Pod) ([][]byte, error) {
```

Comment:
nit: this could be moved to a different package as well

Reply:
Yep, I want to do that as a follow up though. This PR became large enough already.